app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,36 @@
+class VotesController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    vote = current_quote.votes.new vote_params.merge(user: current_user)
+    if vote.save
+      flash[:notice] = "Voted!"
+    else
+      flash[:alert] = "Something went wrong!"
+    end
+    redirect_to quotes_path
+  end
+
+  def update
+    @quote = current_quote
+    vote = @quote.votes.find_by!(user: current_user)
+    flash[:notice] = "Vote updated" if vote.update vote_params
+    redirect_to quotes_path
+  end
+
+  def destroy
+    @vote = current_user.votes.find params[:id]
+    flash[:notice] = "Vote deleted" if @vote.destroy
```

Comment:
I don't think destroy could fail without raising an error?  I had had elses in this and update should still have one now I'm realizing.  I think I messed this up when I was trying to clean up/refactor my code some before I pushed it again since in the next PR an update with an invalid value will throw an error.